src/socket/raw.rs

```diff
@@ -196,7 +196,11 @@ impl<'a, 'b> RawSocket<'a, 'b> {
                 IpVersion::Ipv4 => {
                     let mut packet = Ipv4Packet::new_checked(buffer.as_mut())?;
                     if packet.protocol() != protocol { return Err(Error::Unaddressable) }
-                    packet.fill_checksum();
+                    if emit_checksum.ipv4.tx() {
+                        packet.fill_checksum();
+                    } else {
+                        packet.set_checksum(0);
```

Comment:
What's the rationale here?

Reply:
Ensuring the checksum field is surely zeroed in the resulting packet.
Not sure if the implementation has an issue for other cases,
but it did cause issues for ICMPV4 for STM32, as indicated by the reference manual:

```
Note that: for ICMP-over-IPv4 packets, the checksum field in the ICMP packet must
always be 0x0000 in both modes, because pseudo-headers are not defined for such
packets. If it does not equal 0x0000, an incorrect checksum may be inserted into the
packet.
```
http://www.st.com/content/ccc/resource/technical/document/reference_manual/51/f7/f3/06/cd/b6/46/ec/CD00225773.pdf/files/CD00225773.pdf/jcr:content/translations/en.CD00225773.pdf (page 846)

Reply:
Okay, this is worth a small comment in every call site.